spec/rest/email_validations_spec.rb

```diff
@@ -126,6 +126,28 @@
         end
 
       end
+
+      context "will refresh data - on consecutive calls" do 
+        let(:incompleted_query) do
+          { "progress"=> { "noOfTotalEntries" => 0, "noOfCompletedEntries" => 1 } }
+        end
+
+        let(:completed_query) do
+          { "progress"=> { "noOfTotalEntries" => 1, "noOfCompletedEntries" => 1 } }
+        end
+
+        it "calls query API on current object if job is not finished in first call" do
+          allow(@email_validations).to receive(:query).and_return(incompleted_query)
+          @email_validations.query
+          expect(@email_validations.instance_variable_set('@response', incompleted_query))
+          expect(@email_validations.completed?).to be false
+
+          allow(@email_validations).to receive(:query).and_return(completed_query)
+          @email_validations.query
+          expect(@email_validations.instance_variable_set('@response', completed_query))
+          expect(@email_validations.completed?).to be true
+        end
```

Comment:
I have tested all above updated specs, but with following update needed 
For ex: 1
```ruby
  expect(JSON).to receive(:parse) 
```
should be 

```ruby
 expect(JSON).to receive(:parse).and_return(incompleted_query)
```
Otherwise it will return `nil` in code.

and similarly ex: 2
```ruby
  expect(JSON).to receive(:parse) 
```
should be 

```ruby
 expect(JSON).to receive(:parse).and_return(completed_query)
```
Pushing these updates shortly.